tests/unit/alp_relations.cpp

```diff
@@ -0,0 +1,345 @@
+
+/*
+ *   Copyright 2021 Huawei Technologies Co., Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <iostream>
+#include <sstream>
+#include <vector>
+
+#include <alp.hpp>
+
+void alp_program( const size_t & n, alp::RC & rc ) {
+
+	(void) n;
+
+	/**
+	 * Basic checks on the less-than relation (lt).
+	 * lt is a strict total order and therefore also a strict partial order.
+	 */
+	typedef alp::relations::lt< double > dbl_lt;
+
+	static_assert( alp::is_relation< dbl_lt >::value );
+
+	static_assert( !( alp::is_partial_order< dbl_lt >::value ) );
+
+	static_assert( alp::is_strict_partial_order< dbl_lt >::value );
+
+	static_assert( !( alp::is_total_order< dbl_lt >::value ) );
+
+	static_assert( alp::is_strict_total_order< dbl_lt >::value );
+
+	static_assert( !( alp::is_equivalence_relation< dbl_lt >::value ) );
+
+	if( !( dbl_lt::check(2.4, 5) ) ) {
```

Comment:
missing spaces within inner-most parentheses. Also in other similar appearances in this file.